DTS-STN/secure-client-hub
# Replace logging with Pino

## [ADO-139692](https://dev.azure.com/VP-BD/DECD/_workitems/edit/139692)

Fixed [AB#139692](https://dev.azure.com/VP-BD/d5ca3cd4-19cb-4c8c-b3e6-a7068f4677e4/_workitems/edit/139692)

### Changelog
fix: Updated existing logging implementation to use Pino, added logging to some other areas

### Description of proposed changes:
This PR updates our logger implementation. Previously we were using a custom implementation but this updates our app to use Pino which is the recommended library in Next.js docs and brings our project in line with what the Next template repo is doing. This PR does not include trace logging in our middleware since the middleware is currently not working, but that will be added in a later PR. The logging in this PR mainly tracks errors that are thrown when fetching data in our `getServerSideProps` and errors that pop up during authentication. This is relatively basic and further logging should added at a later date.

### What to test for/How to test
1. Pull in branch
2. In `my-dashboard.js` add `logger.error("This is an error")` at line 95
3. Type `npm run dev`
4. Navigate to `/my-dashboard`
5. Ensure that you see the error logged in your terminal
